### ISSUE TITLE
template with_dict incorrect behaviour with undefined variables in hash values

### ISSUE THREAD

**Issue by nerzhul** [ansible/ansible#15475]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A
##### SUMMARY

When using template module i found an anoying return from with_dict when a dict value variable wasn't defined:

```
TASK [ringo : CONFIG | Deploy configuration] *******************************
[DEPRECATION WARNING]: Skipping task due to undefined Error, in the future this will be a fatal error.. This feature will be removed in a future release. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
fatal: [172.23.0.1]: FAILED! => {"failed": true, "msg": "with_dict expects a dict"}
```

When i copy the dict with set_fact i have the following, correct, behaviour:

```
TASK [ringo : set_fact] *********************************************************
fatal: [172.23.0.1]: FAILED! => {"failed": true, "msg": "'deployenv' is undefined"}
```
##### STEPS TO REPRODUCE

Execute the following task without definining deployenv variable

```
- name: "CONFIG | Deploy configuration"
  template: src="{{ item.value.s }}/{{ item.key }}" dest="{{ item.value.d }}/{{ item.key }}"
  with_dict:
    'amazing_java.properties':
      s: "/etc/deploy/repository/{{ deployenv }}/ringo/"
      d: "/etc/ringo/"
```
##### EXPECTED RESULTS

```
fatal: [172.23.0.1]: FAILED! => {"failed": true, "msg": "'deployenv' is undefined"}
```
##### ACTUAL RESULTS

```
[DEPRECATION WARNING]: Skipping task due to undefined Error, in the future this will be a fatal error.. This feature will be removed in a future release. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
fatal: [172.23.29.116]: FAILED! => {"failed": true, "msg": "with_dict expects a dict"}
```

**Comment by JeanFred** [comment]
I had this issue in 2.2.0, and it appears fixed in 2.2.1

**Comment by nerzhul** [comment]
Exact
